src/NewButton/button.tsx

```diff
@@ -0,0 +1,220 @@
+        background-color: ${get('colors.btn.danger.hoverBg')({theme})};
+        border-color: ${get('colors.btn.danger.hoverBorder')({theme})};
+        box-shadow: ${get('shadows.btn.danger.hoverShadow')({theme})};
+      }
+      // focus must come before :active so that the active box shadow overrides
+      &:focus:not([disabled]) {
+        border-color: ${get('colors.btn.danger.focusBorder')({theme})};
+        box-shadow: ${get('shadows.btn.danger.focusShadow')({theme})};
+      }
+
+      &:active:not([disabled]) {
+        color: ${get('colors.btn.danger.selectedText')({theme})};
+        background-color: ${get('colors.btn.danger.selectedBg')({theme})};
+        box-shadow: ${get('shadows.btn.danger.selectedShadow')({theme})};
+        border-color: ${get('colors.btn.danger.selectedBorder')({theme})};
+      }
+
+      &:disabled {
+        color: ${get('colors.btn.danger.disabledText')({theme})};
+        background-color: ${get('colors.btn.danger.disabledBg')({theme})};
+        border-color: ${get('colors.btn.danger.disabledBorder')({theme})};
+      }
+    `,
+    invisible: `
+      color: ${get('colors.accent.fg')({theme})};
+      background-color: transparent;
+      border: 0;
+      border-radius: ${get('radii.2')({theme})};
+      box-shadow: none;
+    
+      &:disabled {
+        color: ${get('colors.primer.fg.disabled')({theme})};
+      }
+      &:focus:not([disabled]) {
+        box-shadow: ${get('shadows.btn.focusShadow')({theme})};
+      }
+      &:hover:not([disabled]) {
+        background-color: ${get('colors.btn.hoverBg')({theme})};
+      }
+      &:active:not([disabled]) {
+        background-color: ${get('colors.btn.selectedBg')({theme})};
+      }
+    `
+  }
+  return style[variant]
+}
+
+const getSizes = (size = 'medium', variant: VariantType = 'default', iconOnly: boolean) => {
+  let paddingTop, paddingLeft
+  switch (size) {
+    case 'small':
+      paddingTop = 3
+      paddingLeft = 12
+      break
+    case 'large':
+      paddingTop = 9
+      paddingLeft = 20
+      break
+    case 'medium':
+    default:
+      paddingTop = 5
+      paddingLeft = 16
+  }
+  if (iconOnly) {
+    paddingLeft = paddingTop + 2
+  }
+  if (variant === 'invisible') {
+    paddingTop = paddingTop + 1
+  }
+  return `
+    padding:${paddingTop}px ${paddingLeft}px;
+  `
+}
+const ButtonBase = styled.button<ButtonBaseProps & {iconOnly: boolean}>`
+  ${buttonBaseStyles}
+  ${props => getVariantStyles(props.theme, props.variant)}
+  ${props => getSizes(props.size, props.variant, props.iconOnly)}
+  ${sizes}
+  ${sx}
+  ${fontSize}
+`
+
+const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ComponentProps<typeof ButtonBase>>(
+  ({children, ...props}, forwardedRef) => {
+    const {icon: Icon, caret, size} = props
+    let iconOnly = false
+    if (Icon && !children) {
+      iconOnly = true
+    }
+    const iconWrapStyles = {
+      display: 'inline-block',
```

Comment:
The CSS would need some adjusting with my recommendation for icon handling. I could see CSS grid being used here to to handle slots, similar to what I'm doing with [Action List](https://github.com/primer/css/pull/1657/files#diff-1012f63983e65b582649d97a326609627c08f3ae717a127979884e3955a23151R291). I'm using grid-area to create slots, and allowing children to fill them- and if a child exists, `margin-right`. If you want to explore grid together I'm happy to pair on it!

Reply:
Would love to pair on this!

Reply:
This was a great session @langermank. you are such a wizard with css and esp grid!